idb/postgres/postgres_integration_test.go

```diff
@@ -1540,13 +1540,23 @@ func TestSearchForInnerTransactionReturnsRootTransaction(t *testing.T) {
 	rootTxid := appCall.Txn.ID()
 
 	err = pgutil.TxWithRetry(pdb, serializable, func(tx pgx.Tx) error {
-		w, err := writer.MakeWriter(tx)
-		require.NoError(t, err)
+		err := writer.AddTransactions(&block, block.Payset, tx)
+		if err != nil {
+			return err
+		}
 
-		err = w.AddBlock(&block, block.Payset, ledgercore.StateDelta{})
-		require.NoError(t, err)
+		err = writer.AddTransactionParticipation(&block, tx)
+		if err != nil {
+			return err
+		}
+
+		w, err := writer.MakeWriter(tx)
+		if err != nil {
+			return err
+		}
+		defer w.Close()
 
-		return nil
+		return w.AddBlock(&block, block.Payset, ledgercore.StateDelta{})
```

Comment:
Can this can use db.AddBlock instead of w.AddBlock to simplify the test?

Reply:
it is your test, why did you use writer? :)

Reply:
I was probably trying to reduce the scope of the unit test, but now that it requires more implementation details from db.AddBlock that makes less sense.

Could you switch it to db.AddBlock?

Reply:
done